api/apps/channels/model.coffee

```diff
@@ -29,20 +29,25 @@ querySchema = (->
   limit: @number().max(Number API_MAX).default(Number API_PAGE_SIZE)
   offset: @number()
   user_id: @objectId()
+  q: @string()
 ).call Joi
 
 #
 # Retrieval
 #
 @find = (id, callback) ->
-  query = if ObjectId.isValid(id) then { _id: ObjectId(id) }
-  db.channels.findOne query, callback
+  return callback 'Invalid channel id' unless ObjectId.isValid(id)
```

Comment:
This [should be wrapped](http://www.devthought.com/2011/12/22/a-string-is-not-an-error/) in an error object `new Error('Invalid channel id')`.